### PR TITLE
Mjb 95522266

### PR DESCRIPTION
**Pull request by cbitstech**
updates to session logic to ingest redcap as well as additional views to support it
also adds a cordova serve function so that building isn’t required before testing

### CODE REVIEW

File: src/partials/sessions.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<h2>Session</h2>
+<div ng-repeat="item in sessions.content" >
+	<div ng-show="item.branching_logic == '' && sessions.currentIndex == $index">
```
Comment (michaelwehrley) [inline src/partials/sessions.html]: Since you are comparing strings, could you use `===`?

File: src/partials/sessions.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    </label>
+  </div>
+  <div ng-if="item.type == 'checkbox'" class="{{item.type}}" ng-repeat="choice in item.choices">
+    <label class='btn btn-info col-md-12 col-sm-12 col-xs-12' style="margin-bottom:5px; text-align:left;">
+      &nbsp;<input type="{{item.type}}" ng-model='responses[item.variable]($index)' value='1'> {{choice}} 
+    </label>
+  </div>
+</div>
+
+</div><br/><br/>
```
Comment (michaelwehrley) [inline src/partials/sessions.html]: Do you need 2 of these `<br/>` tags?  For breaking, you could also use `<div class="clearfix"></div>`.

File: package.json [hunk 3]
```diff
@@ -24,9 +24,9 @@
     "eslint": "^0.19.0",
     "font-awesome": "^4.3.0",
     "mocha": "^2.2.1",
-    "moment": "^2.10.2",
```
Comment (ericcf) [inline package.json]: Won't removing this dependency break some functionality, or is moment included elsewhere?

File: src/js/content/content.js [hunk 4]
```diff
@@ -0,0 +1 @@
+var sessionContent = {};
```
Comment (ericcf) [inline src/js/content/content.js]: This should be injected as an angular constant, not as a global.

File: src/js/controllers/SessionsController.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          redCapJSONContent[key].choices= redCapJSONContent[key].choices.split(" | ");
+      });
+
+      return redCapJSONContent
+
+    };
+
+    function revealLogic(redCapJSONContent){
+      angular.forEach(redCapJSONContent, function(value, key) {
+        debugger;
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: debugger

File: src/js/controllers/SessionsController.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+
+    }
+
+    this.pageResponses = function(redCapJSONContent){
+
+    };
+
+    debugger;
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: debugger

File: src/js/controllers/SessionsController.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+      return revealLogic(splitChoices(redCapJSONContent));
+
+    };
+
+    this.content = parseRedCap(this.redCapJSONContent);
+
+    this.currentIndex = 0;
+
+    this.next = function(currentIndex){
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: This method doesn't do anything.

File: src/js/controllers/SessionsController.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+
+
+
+    }
+
+    this.pageResponses = function(redCapJSONContent){
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: This method doesn't do anything.

File: src/js/controllers/SessionsController.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    this.parseRedCap = {};
+
+    function splitChoices(redCapJSONContent){
+      angular.forEach(redCapJSONContent, function(value, key) {
+          redCapJSONContent[key].choices= redCapJSONContent[key].choices.split(" | ");
+      });
+
+      return redCapJSONContent
+
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: Is there a reason for all the whitespace?

File: src/partials/sessions.html [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<h2>Session</h2>
+<div ng-repeat="item in sessions.content" >
+	<div ng-show="item.branching_logic == '' && sessions.currentIndex == $index">
+	<div class="question well" >
+		{{item.label}}
+	</div>
+	<div ng-if="item.type == 'radio'" class="{{item.type}}" ng-repeat="choice in item.choices">
+    <label class='btn btn-info col-md-12 col-sm-12 col-xs-12' style="margin-bottom:5px; text-align:left;">
+      &nbsp;<input type="{{item.type}}" ng-model='responses[item.variable]' value='{$index+1}'> {{choice}}
```
Comment (ericcf) [inline src/partials/sessions.html]: Inconsistent `'` and `"`.

Comment (ericcf) [inline src/partials/sessions.html]: Where is the `responses` variable declared?

File: src/partials/sessions.html [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  </div>
+  <div ng-if="item.type == 'checkbox'" class="{{item.type}}" ng-repeat="choice in item.choices">
+    <label class='btn btn-info col-md-12 col-sm-12 col-xs-12' style="margin-bottom:5px; text-align:left;">
+      &nbsp;<input type="{{item.type}}" ng-model='responses[item.variable]($index)' value='1'> {{choice}} 
+    </label>
+  </div>
+</div>
+
+</div><br/><br/>
+<button class="btn btn-primary col-md-12 col-sm-12 col-xs-12" ng-click="sessions.next($index)">Continue</button>
```
Comment (ericcf) [inline src/partials/sessions.html]: Is `$index` still available outside of the `ng-repeat` scope?

File: src/js/controllers/SessionsController.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
+(function() {
```
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: This controller needs unit tests.

File: src/partials/sessions.html (hunk 1 above)
Comment (cbitstech) [inline src/partials/sessions.html]: sure!

File: src/partials/sessions.html (hunk 2 above)
Comment (cbitstech) [inline src/partials/sessions.html]: Will add.

File: package.json (hunk 3 above)
Comment (cbitstech) [inline package.json]: Included above--I was clearing up a warning in the npm install chain.

File: src/js/content/content.js (hunk 4 above)
Comment (cbitstech) [inline src/js/content/content.js]: will do.

File: src/js/controllers/SessionsController.js (hunk 5 above)
Comment (cbitstech) [inline src/js/controllers/SessionsController.js]: removing

File: src/js/controllers/SessionsController.js (hunk 6 above)
Comment (cbitstech) [inline src/js/controllers/SessionsController.js]: removing

File: src/js/controllers/SessionsController.js (hunk 7 above)
Comment (cbitstech) [inline src/js/controllers/SessionsController.js]: it's scaffolding for the next commit, but will remove.

File: src/partials/sessions.html (hunk 10 above)
Comment (cbitstech) [inline src/partials/sessions.html]: clarifying with schlange

File: src/partials/sessions.html (hunk 11 above)
Comment (cbitstech) [inline src/partials/sessions.html]: no

File: src/js/controllers/SessionsController.js (hunk 12 above)
Comment (cbitstech) [inline src/js/controllers/SessionsController.js]: aye aye.

File: src/js/controllers/SessionsController.js (hunk 7 above)
Comment (ericcf) [inline src/js/controllers/SessionsController.js]: I suggest that scaffolding be removed before making a pull request. Ideally, commits should be squashed with clear descriptions and code should be as clean as possible.

**Pull request closed without merging**